### PR TITLE
Display rate limit error on 429 GW response.

### PR DESCRIPTION
**Pull request by GhenadieVP**
Identify and display the 429 bad HTTP response code.

https://github.com/radixdlt/babylon-wallet-ios/assets/118184705/13101e3e-2137-44e6-bb61-188856297702

### CODE REVIEW

**Comment by kugel3** [comment]
We should not have any changes to `Localizable.strings` in normal PRs, and ideally not to the generate one either. Something must have gone wrong with rebasing!

**Comment by Sajjon** [comment]
I dont understand, the video did not display any message?

**Comment by GhenadieVP** [comment]
> I dont understand, the video did not display any message?
> 
> Or was that the point? That we USED to show an error, but now we dont? (have not looked at the code, looking now...)
> 
> But your PR description said that we DID want to show a message? So I'm confused :D

Wrong video, sorry for confusion :)

**Review by CyonAlexRDX — APPROVED** [review]
LGTM

File: RadixWallet/Core/SharedModels/HTTPURLSessionResponse/HTTPURLSessionResponse.swift [hunk 1]
```diff
@@ -4,14 +4,23 @@ public struct ExpectedHTTPURLResponse: Swift.Error {
 }
 
 // MARK: - BadHTTPResponseCode
-public struct BadHTTPResponseCode: Swift.Error {
+public struct BadHTTPResponseCode: LocalizedError {
```
[the remark this answers is not in the document]
Comment (GhenadieVP) [reply RadixWallet/Core/SharedModels/HTTPURLSessionResponse/HTTPURLSessionResponse.swift:7]: 1. It is thrown.
2. In basic context of home screen it is thrown, also when loading in Assets view. But I guess, to be sure, I could explicitly always show the 429 related error.

File: RadixWallet/Core/Resources/Generated/L10n.generated.swift [hunk 2]
```diff
@@ -264,8 +264,8 @@ public enum L10n {
       public static let addAnAssetAllow = L10n.tr("Localizable", "accountSettings_specificAssetsDeposits_addAnAssetAllow", fallback: "Allow Deposits")
       /// Add Asset
       public static let addAnAssetButton = L10n.tr("Localizable", "accountSettings_specificAssetsDeposits_addAnAssetButton", fallback: "Add Asset")
-      /// Deny Deposit
-      public static let addAnAssetDeny = L10n.tr("Localizable", "accountSettings_specificAssetsDeposits_addAnAssetDeny", fallback: "Deny Deposit")
+      /// Deny Deposits
+      public static let addAnAssetDeny = L10n.tr("Localizable", "accountSettings_specificAssetsDeposits_addAnAssetDeny", fallback: "Deny Deposits")
```
Comment (kugel3) [inline RadixWallet/Core/Resources/Generated/L10n.generated.swift:268]: How can this still be here even after you rebased? It's already in main.

Comment (kugel3) [reply RadixWallet/Core/Resources/Generated/L10n.generated.swift:268]: Ah, you never ran swiftgen

Comment (kugel3) [reply RadixWallet/Core/Resources/Generated/L10n.generated.swift:268]: On the crowdin PR. Or whoever it was that merged that

File: RadixWallet/Core/SharedModels/HTTPURLSessionResponse/HTTPURLSessionResponse.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (GhenadieVP) [reply RadixWallet/Core/SharedModels/HTTPURLSessionResponse/HTTPURLSessionResponse.swift:7]: For now we will have the error properly shown in "some" contexts. 
I will look later down the line on how to handle the scenario of `never caught in any execution path`, as those paths can be many and will increase with time, we will need some kind of strategy to know when some errors can be silenced or converted to other errors, and when not.

**Pull request merged**